api/sanitize.go

```diff
@@ -0,0 +1,59 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/microcosm-cc/bluemonday"
+)
+
+var sanitizer *bluemonday.Policy
+
+func init() {
+	sanitizer = bluemonday.UGCPolicy()
+}
+
+func SanitizeJSON(s []byte) ([]byte, error) {
+	d := json.NewDecoder(bytes.NewReader(s))
+	d.UseNumber()
+
+	var i interface{}
+	err := d.Decode(&i)
+	if err != nil {
+		return nil, err
+	}
+	sanitize(i)
+	return json.MarshalIndent(i, "", "    ")
+}
+
+func sanitize(data interface{}) {
```

Comment:
Since this accepts all types but doesn't work with all types I'd think returning an error for unhanded types would be nice.  If we can't get compile-time integrity we could at least run-time.